ENIGMAsystem/SHELL/Widget_Systems/xlib/kdialog.cpp

```diff
@@ -132,7 +132,7 @@ static int show_message_helperfunc(string message) {
     str_cancel = string("--yesno \"") + add_escaping(message, false, "") + string("\" --yes-label Ok --no-label Cancel ");
 
   str_command = string("kdialog ") +
-  string("--attach=") + std::to_string((int)enigma::x11::win) + string(" ") + str_cancel +
+  string("--attach=") + std::to_string((unsigned long)enigma::x11::win) + string(" ") + str_cancel +
```

Comment:
This is probably a good place to use your new `window_identifier` call instead of directly accessing window primitives.

Reply:
yee, will do there, son!